onmt/modules/ConvAttention.py

```diff
@@ -0,0 +1,43 @@
+"""
+Conv attention takes a matrix, a query vector and a value vector.
+calculate attention weight by query vector and sum on value vector.
+"""
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+scale_weight = 0.5 ** 0.5
+
+
+def seq_linear(linear, x):
+    batch, hidden_size, length, _ = x.size()
+    h = linear(torch.transpose(x, 1, 2).contiguous().view(
+        batch * length, hidden_size))
+    return torch.transpose(h.view(batch, length, hidden_size, 1), 1, 2)
+
+
+class ConvAttention(nn.Module):
+    def __init__(self, input_size):
+        super(ConvAttention, self).__init__()
+        self.linear_in = nn.Linear(input_size, input_size)
+        self.mask = None
+
+    def applyMask(self, mask):
+        self.mask = mask
+
+    def forward(self, base_target_emb, input, encoder_out_t, encoder_out_c):
```

Comment:
Can you add `aeq` size checks here like in MultiHeadedAttention. It is very helpful for documentation.

Reply:
Can you also make these variable names more readable.

Reply:
According to paper, this `ConvAttention` is also a kind of global attention, so I think it should be incorporated into `GlobalAttention.py`?  As we will implement local attention in the future, I figure dividing various attention mechanisms into these two orthogonal categories would make more sense, and facilitate later use. Also, I suggest name this to `ConvMultiStepAttention`. ConvAttention is too general.

Reply:
`aeq` done!
I have added comment for these variables. If you think necessary, I can modify `encoder_out_t` to `query`, and modify `encoder_out_c` as `value`. In my simple thinking, `encoder_out_t` means encoder_out_top, `encoder_out_c` means encoder_out_combination.
@JianyuZhan `ConvMultiStepAttention` is surely a more reasonable name. How about insert `ConvMultiStepAttention` into `GlobalAttention.py`, I agree with this thinking.